### PR TITLE
feat(934170): new rule to cover IN2OZO3N

### LINKED ISSUE #2827: Add support for PHP `data` url scheme (BB IN2OZO3N)
### The problem

Per BB finding, we are not covering the `data://` url scheme properly.

### Context

- The scheme is mainly used by PHP. PHP supports `data:` scheme without `"//"` like `data:text/plain`, not `data://text/plain`.
- URL scheme comes from http://www.faqs.org/rfcs/rfc2397.html
- A data URL scheme's media type specification can include other parameters; for example, one might specify a charset parameter: `data:text/plain;charset=iso-8859-7,%be%fg%be`

### Proposed solution

- Create new rule to handle `data:` url scheme
- Duplicate the data rules needed until the `include` preprocessor is fixed to handle this case
- Add tests that cover solution and BB finding

### References

- https://github.com/coreruleset/coreruleset/pull/2792

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

This new rule covers the `data:` (without `//`) scheme from PHP.

Needed to modify also included charset rule because it contained anchors that didn't help to match.

Fixes #2827.

### CODE REVIEW

File: regex-assembly/934170.ra [hunk 1]
```diff
@@ -0,0 +1,10 @@
+##! Please refer to the documentation at
+##! https://coreruleset.org/docs/development/regex_assembly/.
+
+##!^ ^
+
+##!> assemble
+data:
```
Comment (theseion) [inline regex-assembly/934170.ra:7]: Run formatting to fix indentation.

**Review by theseion — CHANGES_REQUESTED** [review]

File: regex-assembly/934170.ra (hunk 1 above)
Comment (fzipi) [reply regex-assembly/934170.ra]: Done.

File: tests/regression/tests/REQUEST-934-APPLICATION-ATTACK-GENERIC/934170.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            headers:
+              Host: "localhost"
+              User-Agent: "ModSecurity CRS 3 Tests"
+              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
+              Accept-Encoding: gzip,deflate
+              Accept-Language: en-us,en;q=0.5
+            method: GET
+            uri: "/get?ssrf=data://text/plain;base64,SSBsb3ZlIFBIUAo="
+          output:
+            no_log_contains: id "934170"
```
Comment (theseion) [inline tests/regression/tests/REQUEST-934-APPLICATION-ATTACK-GENERIC/934170.yaml:41]: Shouldn't the linter have complained about the missing new line at EOF?

Comment (fzipi) [reply tests/regression/tests/REQUEST-934-APPLICATION-ATTACK-GENERIC/934170.yaml:41]: ...maybe because it's not missing?

Comment (theseion) [reply tests/regression/tests/REQUEST-934-APPLICATION-ATTACK-GENERIC/934170.yaml:41]: 🤷 :)

**Review by theseion — APPROVED** [review]

**Pull request merged**